python/iceberg/hive/hive_tables.py

```diff
@@ -34,8 +41,45 @@ def __init__(self, conf):
     def new_table_ops(self, conf, database, table):
         return HiveTableOperations(conf, self.get_client(), database, table)
 
-    def drop(self, database, table):
-        raise RuntimeError("Not yet implemented")
+    def drop(self, database, table, purge=False):
+        ops = self.new_table_ops(self.conf, database, table)
+        metadata = ops.current()
+
+        with self.get_client() as open_client:
+            _logger.info("Deleting {database}.{table} from Hive Metastore".format(database=database, table=table))
+            open_client.drop_table(database, table, deleteData=False)
+
+        manifest_lists_to_delete = []
+        manifests_to_delete = itertools.chain()
+
+        if purge:
+            if metadata is not None:
+                for s in metadata.snapshots:
+                    manifests_to_delete = itertools.chain(manifests_to_delete, s.manifests)
+                    if s.manifest_location is not None:
+                        manifest_lists_to_delete.append(s.manifest_location())
```

Comment:
`s.manifest_location` is a property, `s.manifest_location()` should throw an error wen called like that shouldn't it?

Reply:
Yes weird, will check

Reply:
Yea it took me a bit to find the mystery.  In the implementation of the property, manifest_list does  "FileSystemInputFile.location" which is itself a method.  Changed the implementation of the property to fix this..

Reply:
Spent some time to find the mystery.  The property getter was calling "FileSystemInputFile.location" which is a method.  Fixed it on the property itself.